lib/Service/OpenAiAPIService.php

```diff
@@ -624,6 +624,54 @@ public function getGenerationImage(string $hash, int $urlId): ?array {
 		}
 	}
 
+	/**
+	 * @return int
+	 */
+	public function getExpTextProcessingTime(): int {
+		return $this->isUsingOpenAi()
+			? intval($this->config->getAppValue(Application::APP_ID, 'openai_text_generation_time', strval(Application::DEFAULT_OPENAI_TEXT_GENERATION_TIME)))
+			: intval($this->config->getAppValue(Application::APP_ID, 'localai_text_generation_time', strval(Application::DEFAULT_LOCALAI_TEXT_GENERATION_TIME)));
+	}
+
+	/**
+	 * @param int $runtime
+	 * @return void
+	 */
+	public function updateExpTextProcessingTime(int $runtime): void {
+		$oldTime = $this->getExpTextProcessingTime();
+		$newTime = intval((1 - Application::EXPECTED_RUNTIME_LOWPASS_FACTOR) * $oldTime + Application::EXPECTED_RUNTIME_LOWPASS_FACTOR * $runtime);
+
+		if ($this->isUsingOpenAi()) {
+			$this->config->setAppValue(Application::APP_ID, 'openai_text_generation_time', strval($newTime));
+		} else {
+			$this->config->setAppValue(Application::APP_ID, 'localai_text_generation_time', strval($newTime));
```

Comment:
intval seems redundant, `newTime` can directly do `strval(...)`

Reply:
same in `updateExpImgProcessingTime`

Reply:
That's true. I just thought that it'd be nicer to store an integer instead of a float since we don't care much for a second here or there. But I'm fine with taking that out if you think it makes more sense.

Reply:
ah I didn't notice the float to int conversion, you can ignore this or do both operations above itself.

Reply:
> or you could leave it as int even: https://github.com/nextcloud/server/blob/master/lib/public/IConfig.php#L136

Oh, I'd forgotten you could do that! Thanks!